webapp/sources/rudder/rudder-templates/src/main/scala/com/normation/templates/FillTemplatesService.scala

```diff
@@ -148,34 +148,33 @@ class FillTemplatesService {
    * If a content is not in the cache, it will create the StringTemplate instance, and return it
    *
    */
-  private[this] var cache = Map[String, SynchronizedFileTemplate]()
+  private[this] val cache = ZioRuntime.unsafeRun(Ref.make(Map[String, SynchronizedFileTemplate]()))
 
   def clearCache(): IOResult[Unit] = {
     semaphore.withPermit {
-      UIO.effectTotal {
-        cache = Map()
-      }
+      cache.set(Map())
     }
   }
 
   def getTemplateFromContent(templateName: String,content: String): IOResult[SynchronizedFileTemplate] = {
-    for {
-      c <- semaphore.withPermit(cache.get(content) match {
+    semaphore.withPermit(for {
```

Comment:
here, you are locking with the same semaphore than fill ; you should have a distrinct semaphore (if possible)

Reply:
they have the same name but there is two `ZioRuntime.unsafeRun(Semaphore.make(1))` so there are not the same (actually, there is one semaphore for each `SynchronizedFileTemplate`, so for each fill)